kivy/lib/osc/OSC.py

```diff
@@ -268,22 +346,33 @@ def handle(self, data, source = None):
 
     def dispatch(self, message, source = None):
         """Sends decoded OSC data to an appropriate calback"""
+        if not message or len(message) == 0: # ignore empty messages
+            return
         if type(message[0]) == list :
             # smells like nested messages
             for msg in message :
                 self.dispatch(msg, source)
         elif type(message[0]) == str :
+            valid = False
             # got a single message
             try:
                 address = message[0]
                 callbackfunction = self.callbacks[address]
```

Comment:
you can simplify the fallback to defaults with a self.callbacks.get('address', self.callbacks.get('default')) and checking result is not None, you should be able to get ride of both try/except, and of the "valid" variable.